src/librustc_typeck/collect.rs

```diff
@@ -1858,7 +1858,7 @@ fn explicit_predicates_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredicat
                 // Keep the type around in a dummy predicate, in case of no bounds.
                 // That way, `where Ty:` is not a complete noop (see #53696) and `Ty`
                 // is still checked for WF.
-                if bound_pred.bounds.is_empty() {
+                if bound_pred.bounds.is_empty() && !ty.has_escaping_bound_vars() {
```

Comment:
Huh, this was previously changed from a WF predicate, to an outlives one?

I suspect instead of changing the condition here (which would hide e.g. `where for<'a> Foo<'a>:,` from WF), you just need to change the `ty::Binder::dummy` to `ty::Binder::bind`.

Reply:
Thanks! That should work fine.